packages/atomic/src/components/commerce/atomic-commerce-product-list/atomic-commerce-product-list.tsx

```diff
@@ -193,12 +194,48 @@ export class AtomicCommerceProductList
     );
   }
 
-  private getPropsForAtomicProduct(product: Product) {
+  private logWarning(message: string) {
+    this.bindings.engine.logger.warn(message);
+  }
+
+  private getInteractiveProduct(product: Product, index: number) {
+    let parentController: ProductListing | Search;
+    if (this.bindings.interfaceElement.type === 'product-listing') {
+      parentController = buildProductListing(this.bindings.engine);
+    } else {
+      parentController = buildSearch(this.bindings.engine);
+    }
+
+    const {name, price, productId, warning} =
+      ProductTemplatesHelpers.getRequiredProductPropertiesForAnalytics(product);
+
+    const pagination = parentController.pagination();
+    const position =
+      pagination.state.page * pagination.state.pageSize + index + 1;
```

Comment:
Expected position is 1-based (see https://github.com/coveo-platform/analytics_schema/blob/0dc47fa9ad35aa119e7a2f22592e09974e860685/schemas/client/api.event.ec.productClick/1/api.event.ec.productClick.1.request.json#L27)

Also confirmed with @samisayegh that position must account for pagination offset. This will be clarified in the documentation (see [LENS-1900](https://coveord.atlassian.net/browse/LENS-1900))